src/monattack.cpp

```diff
@@ -185,6 +185,94 @@ void mattack::shriek(monster *z, int index)
     sounds::sound(z->pos(), 50, _("a terrible shriek!"));
 }
 
+void mattack::shriek_alert(monster *z, int index)
+{
+    if( !z->can_act() || z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+
+    if(g->u.sees( *z )){
+    add_msg( _("The %s begins shrieking!"), z->name().c_str());
+    }
+
+    int dist;
+    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 15 ||
+        !z->sees( *target ) ) {
+        return;
+    }
+
+    z->moves -= 150;
+    z->reset_special(index); // Reset timer
+    sounds::sound(z->pos(), 120, _("a piercing wail!"));
+    z->add_effect("shrieking", 10);
+
+}
+
+static int posp( int posc ) //Determines orientation of target from the source
+{
+    if( posc > 0 ) {
+        return 1;
+    }
+    if( posc < 0 ) {
+        return -1;
+    } else {
+        return 0;
+    }
+}
+
+void mattack::shriek_stun(monster *z, int index)
+{
+    if( !z->can_act() || !z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+    int dist;
+    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 7 ||
+        !z->sees( *target ) ) {
+        return;
+    }
+
+    z->reset_special(index); // Reset timer
+
+    float dx = posp( target->posx() - z->posx() );
+    float dy = posp( target->posy() - z->posy() ); //Find direction of player
+    int i = 0; int j = 0;
+    int bsize = 0;  //Set range of our boxes
+    for( int range = 0; range < 4; range++ ) {
```

Comment:
This whole formula is really weird.
You could just create one box (the biggest one) and check the angle, like in `creature.cpp::auto_find_hostile_target`.
Something like:
```
int target_angle = g->m.coord_to_angle(z->posx(), z->posy(), target->posx(), target->posy());
int cone_angle = half of the angle of the shriek cone here.
for( the box that encompasses the entire cone here ) {
    int tile_angle = g->m.coord_to_angle(z->posx(), z->posy(), p.x, p.y);
    int diff = abs( target_angle - tile_angle );
    if( diff + cone_angle > 360 || diff < cone_angle ) {
        continue; // skip the target, because it's outside cone
    }
    // affect the target
}
```